link/src/main/java/com/stripe/android/link/ui/verification/VerificationViewModel.kt

```diff
@@ -0,0 +1,201 @@
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.core.strings.resolvableString
+import com.stripe.android.link.LinkScreen
+import com.stripe.android.link.account.LinkAccountManager
+import com.stripe.android.link.analytics.LinkEventsReporter
+import com.stripe.android.link.injection.NativeLinkComponent
+import com.stripe.android.link.model.AccountStatus
+import com.stripe.android.link.model.LinkAccount
+import com.stripe.android.link.ui.ErrorMessage
+import com.stripe.android.link.ui.getErrorMessage
+import com.stripe.android.ui.core.elements.OTPSpec
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+/**
+ * ViewModel that handles user verification confirmation logic.
+ */
+internal class VerificationViewModel @Inject constructor(
+    private val linkAccount: LinkAccount,
+    private val linkAccountManager: LinkAccountManager,
+    private val linkEventsReporter: LinkEventsReporter,
+    private val logger: Logger,
+    private val goBack: () -> Unit,
+    private val navigateAndClearStack: (route: LinkScreen) -> Unit,
+) : ViewModel() {
+
+    private val _viewState = MutableStateFlow(
+        value = VerificationViewState(
+            redactedPhoneNumber = linkAccount.redactedPhoneNumber,
+            email = linkAccount.email,
+            isProcessing = false,
+            requestFocus = true,
+            errorMessage = null,
+            isSendingNewCode = false,
+            didSendNewCode = false
+        )
+    )
+    val viewState: StateFlow<VerificationViewState> = _viewState
+
+    val otpElement = OTPSpec.transform()
+
+    private val otpCode: StateFlow<String?> =
+        otpElement.otpCompleteFlow.stateIn(viewModelScope, SharingStarted.Lazily, null)
+
+    init {
+        setUp()
+    }
+
+    private fun setUp() {
+        if (linkAccount.accountStatus != AccountStatus.VerificationStarted) {
+            startVerification()
+        }
+
+        viewModelScope.launch {
+            otpCode.collect { code ->
+                code?.let { onVerificationCodeEntered(code) }
+            }
+        }
+    }
+
+    suspend fun onVerificationCodeEntered(code: String) {
+        updateViewState {
+            it.copy(
+                isProcessing = true,
+                errorMessage = null,
+            )
+        }
+
+        linkAccountManager.confirmVerification(code).fold(
+            onSuccess = {
+                updateViewState {
+                    it.copy(isProcessing = false)
+                }
+                navigateAndClearStack(LinkScreen.Wallet)
+            },
+            onFailure = {
+                for (i in 0 until otpElement.controller.otpLength) {
+                    otpElement.controller.onValueChanged(i, "")
+                }
```

Comment:
Can you use otpElement.controller.reset here instead of needing to call onValueChanged for each index? It looks like the reset function is doing this same thing